default/AI/MilitaryAI.py

```diff
@@ -16,6 +17,34 @@
 totMilRating = 0
 num_milships = 0
 verbose_mil_reporting = False
+best_ship_rating_cache = {}  # indexed by turn, value is rating of that turn
+
+
+def cur_best_mil_ship_rating(include_designs=False):
+    """Find the best military ship we have available in this turn and return its rating.
```

Comment:
Ok, let's have some discussion on this. Here is my opinion and reasoning:

1) MilitaryPriority should exclusively consider the designs (should be done in this PR). The reason for this is that stuff like Dragon Tooth from ruins could totally mess up the perception of the strength of newer ships. 

2) The fact that we can or can not build a strong ship is not a really relevant consideration if there are enemies around us *right now*. If we have access to that super strong Scattered Hull ship which totally destroys the enemies around us - who does really care if it takes 15 turns? The way I read the code is that the current threat assessment mostly is for relatively imminent actions (should I go this way, can I invade that planet?) - for that purpose the currently best ship seems to be a much better representation of relative strength to me as these are the ones that define what we can do in the imminent turns.